From htejun@gmail.com  Tue Jun 12 09:21:21 2007
From: Tejun Heo <htejun@gmail.com>
Date: Mon, 11 Jun 2007 14:03:27 +0900
Subject: sysfs: fix condition check in sysfs_drop_dentry()
To: <greg@kroah.com>, <akpm@linux-foundation.org>, <cebbert@redhat.com>, <sandeen@redhat.com>, <maneesh@in.ibm.com>, <cs@tequila.co.jp>
Message-ID: <20070611050327.GL29122@htj.dyndns.org>
Content-Disposition: inline

From: Tejun Heo <htejun@gmail.com>

The condition check doesn't make much sense as it basically always
succeeds.  This causes NULL dereferencing on certain cases.  It seems
that parentheses are put in the wrong place.  Fix it.

Signed-off-by: Tejun Heo <htejun@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/inode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/sysfs/inode.c
+++ b/fs/sysfs/inode.c
@@ -252,7 +252,7 @@ void sysfs_drop_dentry(struct sysfs_dire
 	if (dentry) {
 		spin_lock(&dcache_lock);
 		spin_lock(&dentry->d_lock);
-		if (!(d_unhashed(dentry) && dentry->d_inode)) {
+		if (!d_unhashed(dentry) && dentry->d_inode) {
 			inode = dentry->d_inode;
 			spin_lock(&inode->i_lock);
 			__iget(inode);
